### PR TITLE
End to end testing workflow

### LINKED ISSUE #405: [REQUEST] Add functional tests to testing suite
**Is your feature request related to a problem? Please describe.**
As a local user, I would appreciate end-to-end functional tests be included in *repository* testing suite (as opposed to in a GitHub-triggered-action) in order to make sure the code *in my environment* is working appropriately from end-to-end. 

**Describe the solution you'd like**
A set of end-to-end functional tests for a variety of feed sources and settings run within local tests. 
https://guides.gradle.org/testing-gradle-plugins/ has some suggestions. 

**Describe alternatives you've considered**
While the GitHub triggered functional test is useful, it is not easily run locally by non-contributors *and* isn't necessarily run as a matter of course on local machines prior to pushing code.

### LINKED ISSUE #406: [REQUEST] Add additional feeds for end-to-end, functional tests
**Is your feature request related to a problem? Please describe.**
I've been frustrated when the validator fails and is brittle to different feeds other than MBTA. 

**Describe the solution you'd like**
~5sh additional feeds of varying complexity/quality added to the validator's testing suite  (see issue #405) and continuous integration process. 

**Describe alternatives you've considered**
Running them myself and submitting bugs for each, which isn't ideal b/c developers would ideally catch these bugs before users.

### PR DESCRIPTION
**Pull request by f8full**
closes #406  

**Summary:**

Add a worflow to run the validator on multiple datasets when code is merged on master

**Expected behavior:** 

On future merge to master, the validator should run on both MBTA and STM datasets

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: .github/workflows/endtoend.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: End to end
```
Comment (lionel-nj) [inline .github/workflows/endtoend.yml:1]: Should we add a precision: `End to end testing workflow`?

**Review by lionel-nj — APPROVED** [review]
One in-line comment but generally LGTM .

**Comment by e-lo** [comment]
Thank for working on this. A few questions:

1. Is there a reason why you wouldn't add a list of feeds and then iterate through them?
2. Are you limited in compute time for tests? Especially if it is only running when merging to master it would seem appropriate to add many more feeds using a variety of techniques to check for brittleness across multiple dimensions:  running from local data, from a variety of common feed aggregators, using a handful of settings, etc.
3. See #405 for additional suggestions re using tests rather than just relying on workflows which aren't replicable locally.

File: .github/workflows/endtoend.yml (hunk 1 above)
Comment (fabrice-v) [reply .github/workflows/endtoend.yml:1]: I'd arbitrate against for 2 reasons
- the file is located in a folder named workflow
- this name is what ends up in the badges we display in our readme

**Pull request merged**